include/adaflo/sharp_interface_util.h

```diff
@@ -803,116 +757,190 @@ collect_evaluation_points(const Triangulation<dim, spacedim> &     surface_mesh,
 
 
 
-template <int dim, typename VectorType, typename BlockVectorType>
+/**
+ * Compute force vector for sharp-interface method (mixed level set).
+ */
+template <int dim, int spacedim, typename VectorType, typename BlockVectorType>
 void
-compute_force_vector_sharp_interface(const Triangulation<dim - 1, dim> &surface_mesh,
-                                     const Mapping<dim - 1, dim> &      surface_mapping,
-                                     const FiniteElement<dim - 1, dim> &surface_fe,
-                                     const Quadrature<dim - 1> &        surface_quad,
-                                     const Mapping<dim> &               mapping,
-                                     const DoFHandler<dim> &            dof_handler,
-                                     const DoFHandler<dim> &            dof_handler_dim,
-                                     const BlockVectorType &normal_vector_field,
-                                     const VectorType &     curvature_solution,
-                                     VectorType &           force_vector)
+compute_force_vector_sharp_interface(const Triangulation<dim, spacedim> &surface_mesh,
+                                     const Mapping<dim, spacedim> &      surface_mapping,
+                                     const Quadrature<dim> &     surface_quadrature,
+                                     const Mapping<spacedim> &   mapping,
+                                     const DoFHandler<spacedim> &dof_handler,
+                                     const DoFHandler<spacedim> &dof_handler_dim,
+                                     const double                surface_tension,
+                                     const BlockVectorType &     normal_solution,
+                                     const VectorType &          curvature_solution,
+                                     VectorType &                force_vector)
 {
-  // step 1) collect all locally-relevant surface quadrature points (cell,
-  // reference-cell position,
-  //  quadrature weight)
-  const auto [cells, ptrs, weights, points] =
-    collect_evaluation_points(surface_mesh,
-                              surface_mapping,
-                              surface_fe,
-                              surface_quad,
-                              dof_handler.get_triangulation(),
-                              mapping);
-
-  // step 2) loop over all cells and evaluate curvature and normal in the cell-local
-  // quadrature points
-  //   and test with all test functions of the cell
+  using T = double; // type of data to be communicated (only |J|xW)
 
-  AffineConstraints<double> constraints; // TODO: use the right ones
+  const auto integration_points = [&]() {
+    std::vector<Point<spacedim>> integration_points;
```

Comment:
I suggest to simply fill `integration_points` inside a scope as a means of code structuring, rather than the lambda function. Or use a proper function that explicitly specifies the input arguments and not implicitly captures whole content of the surrounding class.

Reply:
Done.